src/Serializer/AbstractCollectionNormalizer.php

```diff
@@ -54,11 +57,28 @@ public function supportsNormalization(mixed $data, string $format = null, array
         return static::FORMAT === $format && is_iterable($data);
     }
 
+    /**
+     * @deprecated since Symfony 6.3, use "getSupportedTypes(?string $format)" instead
+     */
     public function hasCacheableSupportsMethod(): bool
     {
+        trigger_deprecation('symfony/serializer', '6.3', 'The "%s()" method is deprecated, use "getSupportedTypes()" instead.', __METHOD__);
+
         return true;
     }
 
+    public function getSupportedTypes(?string $format)
+    {
+        return match($format) {
+            static::FORMAT => [
+                HalCollectionNormalizer::class => true,
+                HydraCollectionNormalizer::class => true,
+                JsonApiCollectionNormalizer::class => true,
+            ],
+            default => [],
+        };
```

Comment:
I think this must return `['*' => false]`.

The description of the `getSupportedTypes` method indicates `a result cannot be cached when it depends on the context or on the data.` which the `AbstractCollectionNormalizer` does, it verifies that the provided data is iterable.